src/frontend/screens/Library/index.tsx

```diff
@@ -306,13 +306,24 @@ export default React.memo(function Library(): JSX.Element {
   // top section
   const showRecentGames = libraryTopSection.startsWith('recently_played')
 
+  const favouriteGamesList = useMemo(() => {
+    return showHidden
+      ? favouriteGames.list
+      : favouriteGames.list.filter(
+          (game) =>
+            !hiddenGames.list
+              .map((hidden: HiddenGame) => hidden?.appName)
+              .includes(game.appName)
+        )
+  }, [favouriteGames, showHidden, hiddenGames])
```

Comment:
I think this would be more readable if we do:

```ts
const favouriteGamesList = useMemo(() => {
  if (showHidden) {
    return favouritesGames.list
  }

  return ... thefileredlist
}
```

that ternary operator is kinda complex


also, with that change you can move the `hiddenGames.list.map` thing outside of the filter callback, now it's being calculated for every element:

```ts
const favouriteGamesList = useMemo(() => {
  if (showHidden) {
    return favouritesGames.list
  }

  const hiddenAppNames = hiddenGames.list.map((hidden: HiddenGame) => hidden?.appName)
                                                                      // do we need this "?" here?, can this be undefined?

  return favouriteGames.list.filter((game) => !hiddenAppNames.includes(game.appName))
}
```

Reply:
Oh yeah that makes sense, I'll patch it up.